ComfortZoneApp/app/src/main/java/com/example/comfortzone/fragments/MapFragment.java

```diff
@@ -89,7 +83,7 @@ public void addMapMarkers(GoogleMap googleMap, List<WeatherData> weatherData) {
     }
 
     @Override
-    public void onCityListUpdated(List<WeatherData> cityList) {
-
+    public void onCityListUpdated(List<WeatherData> newCityList) {
+        cityList = newCityList;
```

Comment:
I was wondering if this is always necessary since both list has reference to the same object, so what we really just need to do is refresh list or map? to get the reference, you can just pass it in when you create the fragment